core/src/main/java/io/grpc/internal/TransportSet.java

```diff
@@ -106,10 +110,11 @@
   private boolean shutdown;
 
   /**
-   * The future for the transport for new outgoing requests. 'lock' lock must be held when assigning
-   * to activeTransportFuture.
+   * The future for the transport for new outgoing requests.
    */
-  private volatile SettableFuture<ClientTransport> activeTransportFuture;
+  @Nullable
+  @GuardedBy("lock")
+  private SettableFuture<ClientTransport> activeTransportFuture;
```

Comment:
Done. Also merged `createActiveTransportFuture()` into `obtainActiveTransport()`, and added two test cases for shutdown.